### DISCUSSION TITLE
Why does navigation not work?

### DISCUSSION THREAD

**Discussion by eclere** [unoplatform/uno#17894] [Q&A]
Hello,
I cannot navigate from the code behind. I have tried different methods from the documentation. Nothing happens. I do not receive an exception.

My routes:
`routes.Register(
            new RouteMap("", View: views.FindByViewModel<ShellModel>(),
                Nested:
                [
                    new ("Login", View: views.FindByViewModel<LoginModel>()),
                    new ("Dashboard", View: views.FindByViewModel<DashboardModel>()),
                    new ("ObjektTasks", View: views.FindByViewModel<ObjektTasksPageModel>())//,
                    //new ("Second", View: views.FindByViewModel<SecondModel>()),
                ]
            )
        );`

  Try with click handler:

`public sealed partial class ObjektTasksPage : Page
	{
		public ObjektTasksPage()
		{
			this.InitializeComponent();
		}

        public ObjektTasksPageModel? ViewModel => DataContext as ObjektTasksPageModel;

        private async void Button_Click(object sender, RoutedEventArgs e)
        {
            var nav = this.Navigator();
            await nav.NavigateRouteAsync(this, "Dashboard");
        }
    }`

Try with ViewModel:
`public partial record ObjektTasksPageModel
    {
        BoardService _board;
        readonly INavigator _nav;
        int _taskId = -1;

        public ObjektTasksPageModel(
            BoardService authentication,
            INavigator navigator)
        {
            _board = authentication;
            _nav = navigator;
        }

        public IState<TaskContainer> AufgabenContainer => State<TaskContainer>.Async(this,

        async ct => await _board.GetAufgabeAsync(_taskId)

        );

        public async ValueTask DoneTask(CancellationToken ct)
        {

            var result = await _board.PatchTaskAsync(_taskId, ct);
            if (result)
            {
                _ = await _nav.NavigateRouteAsync(this, "Dashboard");
            }
        }
    }`

XAML Button
` <Button Content="Done" Style="{StaticResource FilledButtonStyle}" Command="{Binding DoneTask}"  />`

Can anybody help?

**Comment by morning4coffe-dev** [comment]
Hey @eclere,

since I don't have the whole sample, I wasn't able to reproduce the issue with a similar setup. Can you by any chance please provide full repro project?

If not, you can have a look at my sample where I tried to reproduce this: [UnoNavigationSample.zip](https://github.com/user-attachments/files/16559208/UnoNavigationSample.zip). Make sure to also check if you have the correct views registered in RegisterRoutes and you have the `.UseNavigation(ReactiveViewModelMappings.ViewModelMappings, RegisterRoutes)` set for Configure.

**Reply by eclere** [reply]
Hello,

thank you for the reply. I was able to modify your example with the problem. You can [download ](https://semiq.eu/uno/UnoApp3.zip) it.

Thanks fpor help.

**Reply by morning4coffe-dev** [reply]
Nice! Can you please upload the package directly on GitHub? (just dragging the .zip file to this text field should do), I am not able to download it from the website. Thanks.

**Reply by eclere** [reply]
[UnoApp3.zip](https://github.com/user-attachments/files/16573026/UnoApp3.zip)

No problem;)

**Reply by morning4coffe-dev** [reply]
I see, thank you for providing the repro. To make this work, I think you need to use the `DependsOn` parameter when defining the route. For example, something like this should work when navigating to "Other" from "Entity":

```csharp
routes.Register(
    new RouteMap("", View: views.FindByViewModel<ShellModel>(),
        Nested:
        [
            new ("Main", View: views.FindByViewModel<MainModel>(), IsDefault: true),
            new ("Entity", View: views.FindByViewModel<EntityPageModel>()),
            new ("Other", View: views.FindByViewModel<OtherModel>(), DependsOn: "Entity"),
        ]
    )
);
```

If you specifically need to navigate to "Main" this way, that might be a limitation, and I'm not sure if it would work well. However, we can open an issue in the `uno.extensions` repo to track this further.
